### PR TITLE
handle more than limit users

### PR DESCRIPTION
**Pull request by butonic**
Currently, the user_ldap might return more than 500 users per page when multiple servers or base dns are configured. The AllUsersIterator will simply ignore any results beyond the page size of 500 on every page, causing them to not by synced when running occ.

This has not been noticed for a while because users will still be synced when they log in.

This PR hardens the AllUsersIterator and allows it to handle bigger pages. The SeenUsersIterator is not affected, because it works on the account table.

### CODE REVIEW

File: lib/private/User/Sync/AllUsersIterator.php [hunk 1]
```diff
@@ -35,14 +48,25 @@ public function __construct(UserInterface $backend) {
...
 			$this->page++;
 			$offset = $this->page * self::LIMIT;
 			$this->data = $this->backend->getUsers('', self::LIMIT, $offset);
+			$this->dataPos = 0;
+			$this->endPos = \count($this->data);
+			$this->hasMoreData = $this->endPos >= self::LIMIT;
 		}
 	}
+
+	protected function currentDataPos() {
```
Comment (PVince81) [inline lib/private/User/Sync/AllUsersIterator.php]: seems this is unused, remove ?

**Review by PVince81 — CHANGES_REQUESTED** [review]
looks good, just one minor thing to remove

File: lib/private/User/Sync/AllUsersIterator.php (hunk 1 above)
Comment (butonic) [reply lib/private/User/Sync/AllUsersIterator.php]: nope it overwrites the parent method

**Review by PVince81 — APPROVED** [review]
:+1: please backport

**Pull request merged**